Draft 2 - Add Blank Columns and Draggable Backpack

NBumgardner opened the pull request:
Partially implement draft 2 by adding blank columns, and a draggable backpack which starts at the bottom of the first column.

The backpack uses a placeholder icon, and looks 20% larger while selected for fake-tactile user feedback.

## Screenshots

![pr-2_columns-and-backpack](https://github.com/SamBumgardner/packrat-game/assets/11843918/cb831dab-c297-437c-ba4a-00e665645870)
_**Screenshot 1:** Local screenshot after waiting at least 3 seconds on the gameplay screen._

![pr-2_enlarge](https://github.com/SamBumgardner/packrat-game/assets/11843918/2d70c32c-681c-46ad-b65f-b40c5fe1c7b4)
_**Screenshot 2:** Local screenshot while the backpack is selected._

NBumgardner committed:
Cut Unused Process Methods to Stop Warnings

NBumgardner committed:
Add Numbered Columns, Fail to Rename

NBumgardner committed:
Cut Headers from Columns

NBumgardner committed:
Remove Leftover Column Naming, Add Backpack

NBumgardner committed:
Add Backpack Snapped to First Column

NBumgardner committed:
Fix AnchorPoint to Be Global

NBumgardner committed:
Make Backpack Aware of Being Hovered Over

NBumgardner committed:
Click to Start Dragging or Release Backpack

NBumgardner committed:
Document Recent Scenes, x2

NBumgardner committed:
Cut Dead or Unhelpful Backpack Code

NBumgardner committed:
Start with 3 Centered Columns

NBumgardner committed:
Initialize Backpack At First Column

NBumgardner committed:
Enlarge Backpack 20 Percent While Selected

NBumgardner committed:
Alphabetize and Encapsulate for Readability

SamBumgardner commented on gameplay/Gameplay.gd:
```diff
@@ -6,14 +6,28 @@ extends Control
 
-# Called every frame. 'delta' is the elapsed time since the previous frame.
 func _process(delta):
-	pass
+	if (_are_columns_rendered()):
+		_backpack_initialized = true
+		$Backpack.snap_position_to_column($Columns/GameplayColumn)
+		$Backpack.visible = true
+		
+func _are_columns_rendered():
+	return (
+		not _backpack_initialized
+		and _columns_ready_count == _expected_columns_initialized_count
+	)
+
+func _increment_columns_ready_count():
+	_columns_ready_count += 1
```
This "is it time to snap the backpack to its correct position" check should hopefully be able to work in a different way. 

The point of the _ready method is it should only be called after all child nodes have finished initializing. 

I think I remember that it didn't work well when we were doing initial testing before, but I'm interested in finding a more straightforward way to determine "after columns finished sorting themselves out"

SamBumgardner replied:
I found the answer: the Columns container emits an event (`sort_children`) when it finishes positioning and sizing its contents.

We can use that to snap the backpack to a column whenever the number of columns (or their position, I guess) changes.

NBumgardner replied:
Fixed by merging your PR, thank you for figuring out the right event to listen for.

SamBumgardner committed:
simplify check for when columns load

Previous code required each column to emit an event when they were positioned and owning script had to keep track of when each was completed.

Now we can just wait for the container wrapping the columns to report that everything inside of it has been sorted and resized. Will probably be useful in the future when we can add/subtract columns.

NBumgardner committed:
Separate Purpose from Input Checks

NBumgardner committed:
Merge pull request #3 from SamBumgardner/patch-column-load

simplify check for when columns load
